arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedColumnIterator.java

```diff
@@ -214,20 +214,6 @@ protected int nextBatchOf(
     }
   }
 
-  public class FixedWidthTypeBinaryBatchReader extends BatchReader {
```

Comment:
I don't think we can just remove this. We should add a deprecation message (`@deprecated since 1.7.0, will be removed in 1.8.0`) here and to `fixedWidthTypeBinaryBatchReader`as it's possible that consumers of Iceberg could use this. See also #3249 on how we did the deprecation and #7987 later removed this

Reply:
Ok, I will deprecate rather than remove the unused code.
There was no revapi failure, so I thought the removal was ok.
In any case, it's a useful exercise to show that with the code removed the build and all tests still pass.